drivers/timer/arm_arch_timer.c

```diff
@@ -10,10 +10,9 @@
 #include <spinlock.h>
 #include <arch/cpu.h>
 
-#define CYC_PER_TICK	((uint32_t)((uint64_t)sys_clock_hw_cycles_per_sec() \
-			       / (uint64_t)CONFIG_SYS_CLOCK_TICKS_PER_SEC))
-
-#define MAX_TICKS	((0xffffffffu - CYC_PER_TICK) / CYC_PER_TICK)
+#define CYC_PER_TICK	((uint64_t)sys_clock_hw_cycles_per_sec() \
+			/ (uint64_t)CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+#define MAX_TICKS	INT_MAX
```

Comment:
Please make this `INT32_MAX` since we really do need it the duration it provides to fit in the `int32_t` argument to clock announce.

Reply:
ACK